src/instr/instr_paje_types.cpp

```diff
@@ -45,33 +45,14 @@ static type_t newType (const char *typeNameBuff, const char *key, const char *co
   return ret;
 }
 
-void PJ_type_free (type_t type)
+s_type::~s_type()
 {
-  value* val;
-  char *value_name;
-  xbt_dict_cursor_t cursor = nullptr;
-  xbt_dict_foreach (type->values, cursor, value_name, val) {
-    XBT_DEBUG("free value %s, child of %s", val->name, val->father->name);
-  }
-  xbt_dict_free (&type->values);
-  xbt_free (type->name);
-  xbt_free (type->id);
-  xbt_free (type->color);
-  xbt_dict_free (&type->children);
-  xbt_free (type);
-  type = nullptr;
-}
-
-void recursiveDestroyType (type_t type)
```

Comment:
The son will not be automatically be removed if you remove the father. You have to do it explicitely.